### PR TITLE
Add an option to choose websocket protocol

### PR DESCRIPTION
**Pull request by singiamtel**
Now that wss is the default this change is needed to support insecure servers

Needed the tests change for the pre-commit hook to work 

- **tests: Allow running tests under different account names** 
- **feat: Add server protocol to Client**

### CODE REVIEW

File: client.d.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ type ClientOpts = {
 	server?: string,
 	serverid?: string,
 	port?: number,
+	serverProtocol: string,
```
Comment (PartMan7) [inline client.d.ts]: Should this be optional?

File: client.js [hunk 2]
```diff
@@ -122,7 +123,8 @@ class Client extends EventEmitter {
 		const id = ~~(Math.random() * 900) + 100;
 		const chars = 'abcdefghijklmnopqrstuvwxyz0123456789_'.split('');
 		const str = Array.from({ length: 8 }, () => chars[~~(Math.random() * 36)]).join('');
-		const conStr = `wss://${this.opts.server}${this.opts.port ? `:${this.opts.port}` : ''}/showdown/${id}/${str}/websocket`;
+		const conStr = `${this.opts.serverProtocol}://${this.opts.server}\
+${this.opts.port ? `:${this.opts.port}` : ''}/showdown/${id}/${str}/websocket`;
```
Comment (PartMan7) [inline client.js]: Could this all be destructured out on the line before (eg: `const { protocol, port = ':443' } = this.opts`) so it doesn't overflow into multi-line template literal? Having the next line be non-indented is less cute

**Review by PartMan7 — CHANGES_REQUESTED** [review]

File: client.d.ts (hunk 1 above)
Comment (singiamtel) [reply client.d.ts]: Oops, that was an oversight. Fixed

File: client.js (hunk 2 above)
Comment (singiamtel) [reply client.js]: Did

**Comment by PartMan7** [comment]
Could you also update the docs in the README for this?

**Review by PartMan7 — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -79,6 +79,7 @@ type options = {
 	password?: string, // The password for the username you're connecting to. Leave this blank if the account is unregistered.
 	server?: string, // The server to which you wish to connect to - defaults to 'sim3.psim.us'.
 	port?: number, // The port on which you're connecting to. Can also be specified in server as `url:port`, in which case leave this field blank.
+	serverProtocol?: number, // The protocol used for the websocket connection. Defaults to wss, but can be changed to ws (insecure).
```
Comment (PartMan7) [inline README.md]: `'wss' | 'ws' | string` (or just `string`)

File: test/index.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ dotenv.config();
 const { Client, Tools, Data } = require('../client.js');
 
 const Bot = new Client({
-	username: 'PS-Client',
+	username: process.env.USERNAME ?? 'PS-Client',
```
Comment (PartMan7) [inline test/index.js]: Ah, found it; could you make this `process.env.PS_USERNAME` instead?

**Review by PartMan7 — CHANGES_REQUESTED** [review]

**Comment by PartMan7** [comment]
Also I should probably add a `.env.example` now that there's multiple env values, but that can be done at another time.

File: README.md (hunk 3 above)
Comment (singiamtel) [reply README.md]: ... I really need to sleep lol. Apologies

**Comment by PartMan7** [comment]
This _seems_ to be passing locally; not sure why CI is failing for this

Merging for now; will address fixes if needed

**Review by PartMan7 — APPROVED** [review]
Epic; thanks!

**Comment by PartMan7** [comment]
Published in [3.6.0](https://www.npmjs.com/package/ps-client/v/3.6.0)

**Pull request merged**